gravitee-apim-console-webui/src/management/api/proxy/properties/api-properties.component.ts

```diff
@@ -0,0 +1,238 @@
+/*
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Component, Inject, OnDestroy, OnInit } from '@angular/core';
+import { EMPTY, Subject } from 'rxjs';
+import { StateParams } from '@uirouter/angularjs';
+import { catchError, switchMap, takeUntil, tap } from 'rxjs/operators';
+import { isEmpty, omit, uniqueId } from 'lodash';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+
+import { UIRouterStateParams } from '../../../../ajs-upgraded-providers';
+import { GioTableWrapperFilters } from '../../../../shared/components/gio-table-wrapper/gio-table-wrapper.component';
+import { Property } from '../../../../entities/management-api-v2';
+import { ApiV2Service } from '../../../../services-ngx/api-v2.service';
+import { gioTableFilterCollection } from '../../../../shared/components/gio-table-wrapper/gio-table-wrapper.util';
+import { SnackBarService } from '../../../../services-ngx/snack-bar.service';
+
+type TableDataSource = {
+  _id: string;
+  key: string;
+  value: string;
+  encrypted: boolean;
+  encryptable: boolean;
+};
+
+@Component({
+  selector: 'api-properties',
+  template: require('./api-properties.component.html'),
+  styles: [require('./api-properties.component.scss')],
+})
+export class ApiPropertiesComponent implements OnInit, OnDestroy {
+  private unsubscribe$ = new Subject<void>();
+
+  public isReadOnly = false;
```

Comment:
Don't forget to compute this value according to origin (Management vs. Kubernetes) and user permissions.

Reply:
of course, that why i keep it here ^^